### ISSUE TITLE
Santa .plist not working

### ISSUE THREAD

**Issue by maxinomnom** [zentralopensource/zentral#99]
Hello,

I would like to enroll a device for using Santa.
For the enrollment I can download a plist file and a mobileconfig file from Zentral.

We are using VMWare WSOne (Airwatch) as an MDM where I can't upload any mobileconfig files, just plist payloads.

If I deploy the plist, Santa does not get the configuration and can't connect to the sync server.
If I manually install the mobileconfig on the device it works.
I tried to convert the mobileconfig to a plist file, which worked. If I deploy this plist payload via WSOne, Santa will not get the configuration again...

Is there a way, to give Santa the configuration via plist and not via mobileconfig?

**Comment by np5** [comment]
Hi,

Santa reads the configuration from the [`com.google.santa` preference domain](https://santa.readthedocs.io/en/latest/deployment/configuration/#example-configuration-profile).

Can you compare the configuration profile distributed by WSOne with the one we provide ?

**Comment by maxinomnom** [comment]
Thank you for your fast answer!
Zentral provided the following plist:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>BannedBlockMessage</key>
	<string>The following application has been blocked from executing&lt;br/&gt;
because it has been deemed malicious.</string>
	<key>ClientMode</key>
	<integer>1</integer>
	<key>EnablePageZeroProtection</key>
	<true/>
	<key>ModeNotificationLockdown</key>
	<string>Switching into Lockdown mode</string>
	<key>ModeNotificationMonitor</key>
	<string>Switching into Monitor mode</string>
	<key>SyncBaseURL</key>
	<string>SyncURL-removed</string>
	<key>UnknownBlockMessage</key>
	<string>The following application has been blocked from executing&lt;br/&gt;
because its trustworthiness cannot be determined.</string>
</dict>
</plist>
```

This does not work with the error, that the Payload UUID is missing.

So I tried to convert the mobileconfig into a plist and used this:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>PayloadContent</key>
	<array>
		<dict>
			<key>PayloadContent</key>
			<dict>
				<key>com.google.santa</key>
				<dict>
					<key>Forced</key>
					<array>
						<dict>
							<key>mcx_preference_settings</key>
							<dict>
								<key>BannedBlockMessage</key>
								<string>The following application has been blocked from executing&lt;br/&gt;
because it has been deemed malicious.</string>
								<key>ClientMode</key>
								<integer>1</integer>
								<key>EnablePageZeroProtection</key>
								<true/>
								<key>ModeNotificationLockdown</key>
								<string>Switching into Lockdown mode</string>
								<key>ModeNotificationMonitor</key>
								<string>Switching into Monitor mode</string>
								<key>SyncBaseURL</key>
								<string>SyncURL-removed</string>
								<key>UnknownBlockMessage</key>
								<string>The following application has been blocked from executing&lt;br/&gt;
because its trustworthiness cannot be determined.</string>
							</dict>
						</dict>
					</array>
				</dict>
			</dict>
			<key>PayloadEnabled</key>
			<true/>
			<key>PayloadIdentifier</key>
			<string>URL-removed.zentraltest.santa_configuration</string>
			<key>PayloadType</key>
			<string>com.apple.ManagedClient.preferences</string>
			<key>PayloadUUID</key>
			<string>26192bde-aa74-443a-a72d-f86d36cc1272</string>
			<key>PayloadVersion</key>
			<integer>1</integer>
		</dict>
	</array>
	<key>PayloadDescription</key>
	<string>Google Santa configuration for Zentral</string>
	<key>PayloadDisplayName</key>
	<string>Zentral - Santa configuration</string>
	<key>PayloadIdentifier</key>
	<string>url-removed.zentraltest.santa_configuration</string>
	<key>PayloadOrganization</key>
	<string>Zentral</string>
	<key>PayloadRemovalDisallowed</key>
	<true/>
	<key>PayloadScope</key>
	<string>System</string>
	<key>PayloadType</key>
	<string>Configuration</string>
	<key>PayloadUUID</key>
	<string>d8c699c4-c3b4-4438-b996-deadb7cf5fc3</string>
	<key>PayloadVersion</key>
	<integer>1</integer>
</dict>
</plist>
```


I can't find the error :/

**Comment by np5** [comment]
Looking at [this blog post](https://maclovin.org/blog-native/2015/airwatch-deploy-custom-mcx-profiles)

It looks like WSOne only needs this part:

```xml
<dict>
  <key>PayloadContent</key>
  <dict>
    <key>com.google.santa</key>
    <dict>
      <key>Forced</key>
      <array>
        <dict>
          <key>mcx_preference_settings</key>
          <dict>
            <key>BannedBlockMessage</key>
            <string>The following application has been blocked from executing&lt;br/&gt;
because it has been deemed malicious.</string>
            <key>ClientMode</key>
            <integer>1</integer>
            <key>EnablePageZeroProtection</key>
            <true/>
            <key>ModeNotificationLockdown</key>
            <string>Switching into Lockdown mode</string>
            <key>ModeNotificationMonitor</key>
            <string>Switching into Monitor mode</string>
            <key>SyncBaseURL</key>
            <string>SyncURL-removed</string>
            <key>UnknownBlockMessage</key>
            <string>The following application has been blocked from executing&lt;br/&gt;
because its trustworthiness cannot be determined.</string>
          </dict>
        </dict>
      </array>
    </dict>
  </dict>
  <key>PayloadEnabled</key>
  <true/>
  <key>PayloadIdentifier</key>
  <string>URL-removed.zentraltest.santa_configuration</string>
  <key>PayloadType</key>
  <string>com.apple.ManagedClient.preferences</string>
  <key>PayloadUUID</key>
  <string>26192bde-aa74-443a-a72d-f86d36cc1272</string>
  <key>PayloadVersion</key>
  <integer>1</integer>
</dict>
```

**Comment by maxinomnom** [comment]
> Looking at [this blog post](https://maclovin.org/blog-native/2015/airwatch-deploy-custom-mcx-profiles)
> 
> It looks like WSOne only needs this part:
> 
> ```
> <dict>
>   <key>PayloadContent</key>
>   <dict>
>     <key>com.google.santa</key>
>     <dict>
... ```

I found this website myself and did not read the part about what to leave out...
Unfortunately this does not work either. I do not get why.

The original .mobileconfig file from Zentral seems to be signed. When opening it with an editor, there are random characters besides the plist information. So it is encrypted in a way.

I managed it by WSOne product provisioning. Deploying the profile locally to the machine and install it bei running a terminal command. That is not the elegant way... At least it is working now...

Thank you for helping me :)
